src/context/GrowerContext.js

```diff
@@ -29,26 +35,65 @@ export const GrowerContext = createContext({
 
 export function GrowerProvider(props) {
   const { orgId, orgList } = useContext(AppContext);
+  const { searchParams } = props;
+
+  const {
+    pageSize: pageSizeParam = undefined,
+    currentPage: currentPageParam = undefined,
+    ...filterParams
+  } = Object.fromEntries(searchParams || []);
+
   const [growers, setGrowers] = useState([]);
-  const [pageSize, setPageSize] = useState(24);
+  const [pageSize, setPageSize] = useState(
+    Number(pageSizeParam) || DEFAULT_PAGE_SIZE
+  );
   const [count, setCount] = useState(null);
-  const [currentPage, setCurrentPage] = useState(0);
+  const [currentPage, setCurrentPage] = useState(
+    Number(currentPageParam) || DEFAULT_CURRENT_PAGE
+  );
   const [filter, setFilter] = useState(
-    new FilterGrower({ organization_id: ALL_ORGANIZATIONS })
+    FilterGrower.fromSearchParams({
+      organization_id: ALL_ORGANIZATIONS,
+      ...filterParams,
+    })
   );
   const [isLoading, setIsLoading] = useState(false);
   const [totalGrowerCount, setTotalGrowerCount] = useState(null);
 
   useEffect(() => {
+    if (searchParams) {
+      handleQuerySearchParams({
+        pageSize: pageSize === DEFAULT_PAGE_SIZE ? undefined : pageSize,
+        currentPage:
+          currentPage === DEFAULT_CURRENT_PAGE ? undefined : currentPage,
+        ...filter.toSearchParams(),
+      });
+    }
+
     const abortController = new AbortController();
     // orgId can be either null or an [] of uuids
     if (orgId !== undefined) {
       load({ signal: abortController.signal });
-      // getCount();
+      // getCount({ signal: abortController.signal });
     }
     return () => abortController.abort();
   }, [filter, pageSize, currentPage, orgId]);
 
+  useEffect(() => {
+    if (!searchParams) {
```

Comment:
Could this be part of the previous useEffect, so that there is only one dealing with searchParams?